### DISCUSSION TITLE
we can't use some options in mpv.conf because they are overcast by values in jellyfinmediaplayer.conf or user settings

### DISCUSSION THREAD

**Discussion by sevenrats** [jellyfin/jellyfin-desktop#543] [General]
and people seem to hate it!
several issues to this effect need to be brought under the same banner and discussed. the reason its a discussion and not a bug report is because its not clear exactly how everybody's needs are going to be met in the context of mpv.conf versus jellyfinmediaplayer.conf versus user settings.
This is a place to discuss your needs or suggestions to that effect.
#441
#474
#361 
#563 
#348

**Comment by kradear** [comment]
Not having options to properly configure the subtitles is the biggest drawback. The client shouldn't overwrite any of the settings in mpv.conf
`sub-border-size / sub-font-size / sub-pos ` are unavailable to the user, as far as I've seen.
`sub-use-margins` is the only one available, which helps with not having the subtitles at the absolute bottom of the screen, down in the black bar, far away from the content. For 16:9 content it doesn't help. Subtitles are lower than any other player's default to start with.
Configurable client settings do not allow for customization of these variables, but the default ones overwrite the ones provided by the user in .conf.

`volume-max` is another variable I would enjoy having available.

These are to me too relevant not to be present. Ideally, all default MPV settings that aren't fully configurable in the client would be configurable from .conf

**Comment by MattMckenzy** [comment]
Just a heads up, I've created a pull request that adds some features that could help with issues like these: https://github.com/jellyfin/jellyfin-media-player/pull/561.

Other than a couple of new ASS subtitle settings, I've also added an input textarea where users can define any mpv configuration that should override them from anywhere else.

Feel free to check out my fork with them until whenever these features make it to the published application!

If anyone needs assistance compiling and installing from my fork, feel free to send me a message. Always happy to help!

**Comment by MattMckenzy** [comment]
After using some custom mpv configuration for a bit, I realized "slang" and "alang" won't work since JMP sets and overrides "sid" and "aid" during media load. 

With that realization, I've been trying to make another couple of settings to force "sid" and "aid" to be set to "auto".

I've got something working well here: https://github.com/MattMckenzy/jellyfin-media-player/tree/force-auto-track. The new settings let me use "alang" and "slang" no problem, but the audio and subtitle selection menu available in the player overlay still shows the default track selected even if that's not the case.

Would anyone have an idea how I could update the current track indices in the player? Having a hard time tracking that down.

**Comment by chp-1960** [comment]
I have big problems getting the AC3, DTS, EAC3 output to be recognized in pass-through on HDMI. The only way to have sound is to disable pass-through in the JMP client. Instead, VLC and MPV, on the other hand, work perfectly in pass-through. HDMI Pass-through work fine in Windows too. 

After reading the posts here, I copied my MPV.conf into the directory /snap/jellyfinmediaplayer/25/.local/share/jellyfinmediaplayer/.
 
But even if these MPV.conf settings are read by JMP, it seems that they are not used
[jellyfinmediaplayer.log](https://github.com/jellyfin/jellyfin-media-player/files/14560850/jellyfinmediaplayer.log)
 as we can see in the jellyfinmediaplayer.log log file:
_2024-03-11 15:58:28.423 [debug] unknown @ 0 - cplayer: Applying profile 'default'...
2024-03-11 15:58:28.423 [debug] unknown @ 0 - cplayer: Setting option 'hwdec' = 'vaapi' (flags = 4)
2024-03-11 15:58:28.424 [debug] unknown @ 0 - cplayer: Setting option 'audio-device' = 'alsa/hdmi:CARD=Generic,DEV=0' (flags = 4)
2024-03-11 15:58:28.424 [debug] unknown @ 0 - cplayer: Setting option 'volume' = '100' (flags = 4)
2024-03-11 15:58:28.424 [debug] unknown @ 0 - cplayer: Setting option 'volume-max' = '150' (flags = 4)
2024-03-11 15:58:28.424 [debug] unknown @ 0 - cplayer: Setting option 'audio-channels' = '5.1,stereo' (flags = 4)
2024-03-11 15:58:28.424 [debug] unknown @ 0 - cplayer: Setting option 'af' = 'surround:chl_out=5.1' (flags = 4)
2024-03-11 15:58:28.424 [debug] unknown @ 0 - cplayer: Setting option 'audio-spdif' = 'ac3,dts,eac3' (flags = 4)
2024-03-11 15:58:28.424 [debug] unknown @ 0 - cplayer: Waiting for scripts..._

FYI, JMP runs on KUbuntu 22.04 LTS and JMP is version 1.9.1.
Any idea how to make AC3, DTS, EAC3 to work in JMP?

**Reply by chp-1960** [reply]
I will reply to myself.
First, sorry. I missed an important remark on the project page "need to disable pipewire or else the client will segfault".

So, I uninstalled pipewire:
        sudo apt remove pipewire-audio-client-libraries pipewire

and rebooted.
Of course, I had to re-configure my Pulse Audio and Client settings in JMP.
And miracle, The tests sounds worked as expected in multi-channel mode.

Hope it will help others.
